### PR TITLE
Document Kafka Source Schema management

### PR DESCRIPTION
**Pull request by eolivelli**
Add documentation about Kafka Source manages Schema information.

This patch documents the improvements implemented with: 
#9448 - KafkaSource - allow to manage Avro Encoded messages
#10002 - KafkaSource - Implement KeyValue support for KafkaBytesSource

### CODE REVIEW

File: site2/docs/io-kafka-source.md [hunk 1]
```diff
@@ -17,19 +17,47 @@ The configuration of the Kafka source connector has the following properties.
...
+| `consumerConfigProperties` | Map| false | " " (empty string) | The consumer configuration properties to be passed to consumers. <br/><br/>**Note: other properties specified in the connector configuration file take precedence over this configuration**. |
 | `keyDeserializationClass` | String|false | org.apache.kafka.common.serialization.StringDeserializer | The deserializer class for Kafka consumers to deserialize keys.<br/> The deserializer is set by a specific implementation of [`KafkaAbstractSource`](https://github.com/apache/pulsar/blob/master/pulsar-io/kafka/src/main/java/org/apache/pulsar/io/kafka/KafkaAbstractSource.java).
 | `valueDeserializationClass` | String|false | org.apache.kafka.common.serialization.ByteArrayDeserializer | The deserializer class for Kafka consumers to deserialize values.
 | `autoOffsetReset` | String | false | "earliest" | The default offset reset policy. |
 
+### Schema Management
+
+This Kafka Source connector applies the Schema to the topic depennding on the data type that is present on the Kafka topic.
```
Comment (Jennifer88huang) [inline site2/docs/io-kafka-source.md:35]: depending

File: site2/docs/io-kafka-source.md [hunk 2]
```diff
@@ -17,19 +17,47 @@ The configuration of the Kafka source connector has the following properties.
...
+We are detecting the data type from the `keyDeserializationClass` and `valueDeserializationClass` configuration parameters.
```
Comment (Jennifer88huang) [inline site2/docs/io-kafka-source.md:36]:
```suggestion
You can detect the data type from the `keyDeserializationClass` and `valueDeserializationClass` configuration parameters.
```

File: site2/docs/io-kafka-source.md [hunk 3]
```diff
@@ -17,19 +17,47 @@ The configuration of the Kafka source connector has the following properties.
...
+
+For instance if the `valueDeserializationClass` is `org.apache.kafka.common.serialization.StringDeserializer` then we set Schema.STRING() as schema type
+on the Pulsar topic.
+
```
Comment (Jennifer88huang) [inline site2/docs/io-kafka-source.md:40]:
```suggestion
If the `valueDeserializationClass` is `org.apache.kafka.common.serialization.StringDeserializer`, you can set Schema.STRING() as schema type on the Pulsar topic.

```

File: site2/docs/io-kafka-source.md [hunk 4]
```diff
@@ -95,7 +123,7 @@ Here is an example of using the Kafka source connecter with the configuration fi
 5. Pull a Pulsar image and start Pulsar standalone.
    
    ```bash
-   $ docker pull apachepulsar/pulsar:2.4.0
+   $ docker pull apachepulsar/pulsar:2.8.0
```
Comment (Jennifer88huang) [inline site2/docs/io-kafka-source.md:126]: If the version should be updated with release, maybe we can use variable
`{{pulsar:version}}`

**Review by Jennifer88huang — COMMENTED** [review]
Good job, @eolivelli 
I have one minor suggestion on the writing: when writing technical docs, try to use second person "you" and present tense. 
For guides, refer to https://developers.google.com/style/person

File: site2/docs/io-kafka-source.md [hunk 5]
```diff
@@ -17,19 +17,47 @@ The configuration of the Kafka source connector has the following properties.
...
+
+This Kafka Source connector applies the Schema to the topic depennding on the data type that is present on the Kafka topic.
+We are detecting the data type from the `keyDeserializationClass` and `valueDeserializationClass` configuration parameters.
+
+For instance if the `valueDeserializationClass` is `org.apache.kafka.common.serialization.StringDeserializer` then we set Schema.STRING() as schema type
+on the Pulsar topic.
+
+If `valueDeserializationClass` is `io.confluent.kafka.serializers.KafkaAvroDeserializer` we download the AVRO schema from the Confluent Schema Registry®
+and we set it properly on the Pulsar topic. In this case you have to set `schema.registry.url` inside of the `consumerConfigProperties` configuration entry
+of the source.
```
Comment (Anonymitaet) [inline site2/docs/io-kafka-source.md:43]: Avoid the first-person pronouns `I` and `we`. Use the second-person pronoun `you` as much as possible.

File: site2/docs/io-kafka-source.md [hunk 6]
```diff
@@ -17,19 +17,47 @@ The configuration of the Kafka source connector has the following properties.
...
+
+When `keyDeserializationClass` has any value that is different from the default value of `org.apache.kafka.common.serialization.StringDeserializer` it means 
+that you do not have a String as a key, in this case the Kafka Source uses the KeyValue schema type with the SEPARATED encoding.
```
Comment (Anonymitaet) [inline site2/docs/io-kafka-source.md:46]: Break up long sentences into short ones can improve readability.

File: site2/docs/io-kafka-source.md [hunk 7]
```diff
@@ -17,19 +17,47 @@ The configuration of the Kafka source connector has the following properties.
...
+This way on Pulsar you have a topic with these properties:
+- Schema: KeyValue schema with SEPARATED encoding
+- Key: the content of key of the Kafka message (base64 encoded)
+- Value: the content of value of the Kafka message
+- KeySchema: the schema, detected from `keyDeserializationClass`
+- ValueSchema: the schema, detected from `valueDeserializationClass`
+
+Topic compaction and partition routing use the Pulsar key, that contains the Kafka key, and so they are driven by the same value that you have on Kafka.
+
+When you are consuming data from the Pulsar topic you can use the KeyValue schema, this way you can decode properly the data.
```
Comment (Anonymitaet) [inline site2/docs/io-kafka-source.md:59]:
```suggestion
When you consume data from Pulsar topics, you can use the `KeyValue` schema. In this way, you can decode the data properly.
```

**Review by Anonymitaet — COMMENTED** [review]
Thanks for your contribution. I've left some comments, pls check overall.

File: site2/docs/io-kafka-source.md (hunk 1 above)
Comment (eolivelli) [reply site2/docs/io-kafka-source.md:35]: done

File: site2/docs/io-kafka-source.md (hunk 4 above)
Comment (eolivelli) [reply site2/docs/io-kafka-source.md:125]: thanks I didn't find the variable

File: site2/docs/io-kafka-source.md (hunk 6 above)
Comment (eolivelli) [reply site2/docs/io-kafka-source.md:45]: done

File: site2/docs/io-kafka-source.md (hunk 5 above)
Comment (eolivelli) [reply site2/docs/io-kafka-source.md]: got it

**Comment by eolivelli** [comment]
@Jennifer88huang @Anonymitaet I have addressed your comments and tested again the rendering on my local env.
PTAL

it is very slow to generate the website, I am using
```
yarn build
yarn start
```

File: site2/docs/io-kafka-source.md [hunk 8]
```diff
@@ -17,19 +17,48 @@ The configuration of the Kafka source connector has the following properties.
...
+| `consumerConfigProperties` | Map| false | " " (empty string) | The consumer configuration properties to be passed to consumers. <br/><br/>**Note: other properties specified in the connector configuration file take precedence over this configuration**. |
 | `keyDeserializationClass` | String|false | org.apache.kafka.common.serialization.StringDeserializer | The deserializer class for Kafka consumers to deserialize keys.<br/> The deserializer is set by a specific implementation of [`KafkaAbstractSource`](https://github.com/apache/pulsar/blob/master/pulsar-io/kafka/src/main/java/org/apache/pulsar/io/kafka/KafkaAbstractSource.java).
 | `valueDeserializationClass` | String|false | org.apache.kafka.common.serialization.ByteArrayDeserializer | The deserializer class for Kafka consumers to deserialize values.
 | `autoOffsetReset` | String | false | "earliest" | The default offset reset policy. |
 
+### Schema Management
+
+This Kafka Source connector applies the Schema to the topic depending on the data type that is present on the Kafka topic.
```
Comment (Anonymitaet) [inline site2/docs/io-kafka-source.md:35]:
```suggestion
This Kafka source connector applies the schema to the topic depending on the data type that is present on the Kafka topic.
```

File: site2/docs/io-kafka-source.md [hunk 9]
```diff
@@ -17,19 +17,48 @@ The configuration of the Kafka source connector has the following properties.
...
+You can detect the data type from the `keyDeserializationClass` and `valueDeserializationClass` configuration parameters.
+
+If the `valueDeserializationClass` is `org.apache.kafka.common.serialization.StringDeserializer`, you can set Schema.STRING() as schema type on the Pulsar topic.
+
+If `valueDeserializationClass` is `io.confluent.kafka.serializers.KafkaAvroDeserializer` Pulsar downloads the AVRO schema from the Confluent Schema Registry®
```
Comment (Anonymitaet) [inline site2/docs/io-kafka-source.md:40]:
```suggestion
If `valueDeserializationClass` is `io.confluent.kafka.serializers.KafkaAvroDeserializer`, Pulsar downloads the AVRO schema from the Confluent Schema Registry®
```

File: site2/docs/io-kafka-source.md [hunk 10]
```diff
@@ -17,19 +17,48 @@ The configuration of the Kafka source connector has the following properties.
...
+and set it properly on the Pulsar topic.
+
+In this case you have to set `schema.registry.url` inside of the `consumerConfigProperties` configuration entry
```
Comment (Anonymitaet) [inline site2/docs/io-kafka-source.md:43]:
```suggestion
In this case, you need to set `schema.registry.url` inside of the `consumerConfigProperties` configuration entry
```

File: site2/docs/io-kafka-source.md [hunk 11]
```diff
@@ -17,19 +17,48 @@ The configuration of the Kafka source connector has the following properties.
...
+of the source.
+
+When `keyDeserializationClass` is not `org.apache.kafka.common.serialization.StringDeserializer` it means
```
Comment (Anonymitaet) [inline site2/docs/io-kafka-source.md:46]:
```suggestion
If `keyDeserializationClass` is not `org.apache.kafka.common.serialization.StringDeserializer`, it means 
```

File: site2/docs/io-kafka-source.md [hunk 12]
```diff
@@ -17,19 +17,48 @@ The configuration of the Kafka source connector has the following properties.
...
+If `valueDeserializationClass` is `io.confluent.kafka.serializers.KafkaAvroDeserializer` Pulsar downloads the AVRO schema from the Confluent Schema Registry®
+and set it properly on the Pulsar topic.
+
+In this case you have to set `schema.registry.url` inside of the `consumerConfigProperties` configuration entry
+of the source.
+
+When `keyDeserializationClass` is not `org.apache.kafka.common.serialization.StringDeserializer` it means 
+that you do not have a String as key and the Kafka Source uses the KeyValue schema type with the SEPARATED encoding.
+
+We are also supporting AVRO format for keys as well.
```
Comment (Anonymitaet) [inline site2/docs/io-kafka-source.md:49]:
```suggestion
Pulsar supports AVRO format for keys.
```

File: site2/docs/io-kafka-source.md [hunk 13]
```diff
@@ -17,19 +17,48 @@ The configuration of the Kafka source connector has the following properties.
...
+
+This way on Pulsar you have a topic with these properties:
```
Comment (Anonymitaet) [inline site2/docs/io-kafka-source.md:51]:
```suggestion
In this case, you can have a Pulsar topic with the following properties:
```

File: site2/docs/io-kafka-source.md [hunk 14]
```diff
@@ -17,19 +17,48 @@ The configuration of the Kafka source connector has the following properties.
...
+- Schema: KeyValue schema with SEPARATED encoding
+- Key: the content of key of the Kafka message (base64 encoded)
+- Value: the content of value of the Kafka message
+- KeySchema: the schema, detected from `keyDeserializationClass`
```
Comment (Anonymitaet) [inline site2/docs/io-kafka-source.md:55]:
```suggestion
- KeySchema: the schema detected from `keyDeserializationClass`
```

File: site2/docs/io-kafka-source.md [hunk 15]
```diff
@@ -17,19 +17,48 @@ The configuration of the Kafka source connector has the following properties.
...
+- ValueSchema: the schema, detected from `valueDeserializationClass`
```
Comment (Anonymitaet) [inline site2/docs/io-kafka-source.md:56]:
```suggestion
- ValueSchema: the schema detected from `valueDeserializationClass`
```

File: site2/docs/io-kafka-source.md [hunk 16]
```diff
@@ -17,19 +17,48 @@ The configuration of the Kafka source connector has the following properties.
...
+
+Topic compaction and partition routing use the Pulsar key, that contains the Kafka key, and so they are driven by the same value that you have on Kafka.
+
+When you consume data from Pulsar topics, you can use the `KeyValue` schema. In this way, you can decode the data properly.
+In case you want to access the raw key you can use the Message#getKeyBytes() API.
```
Comment (Anonymitaet) [inline site2/docs/io-kafka-source.md:61]:
```suggestion
If you want to access the raw key, you can use the `Message#getKeyBytes()` API.
```

**Comment by Anonymitaet** [comment]
> @Jennifer88huang @Anonymitaet I have addressed your comments and tested again the rendering on my local env.
> PTAL
> 
> it is very slow to generate the website, I am using
> 
> ```
> yarn build
> yarn start
> ```

@eolivelli yes it works slow but they are the correct commands. We'd better preview the changes on our local environments and [check the broken links (if any)](https://github.com/apache/pulsar/tree/master/site2#check) before sending a doc PR.

**Comment by eolivelli** [comment]
@Anonymitaet thanks again for your suggestions. I have updated the document

**Review by Jennifer88huang — APPROVED** [review]

**Pull request merged**